langchain/src/retrievers/tests/vectara.int.test.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-process-env */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { test, expect } from "@jest/globals";
+
+import { VectaraRetriever } from "../vectara.js";
+import { VectaraStore } from "../../vectorstores/vectara.js";
+
+/**
+ * Steps to run this test:
+ * 1. Create a Vectara account at https://vectara.com/
+ * 2. Create a corpus
+ * 3. Upload a specific document to the corpus (e.g. the US constitution)
+ * 4. Set the appropriate environment variables (VECTARA_CUSTOMER_ID, VECTARA_CORPUS_ID, VECTARA_API_KEY)
+ * 5. Run the test
+ */
+
+test("VectaraRetriever (with filter)", async () => {
+  const store = new VectaraStore({
+    customer_id: process.env.VECTARA_CUSTOMER_ID ? parseInt(process.env.VECTARA_CUSTOMER_ID) : 0,
+    corpus_id: process.env.VECTARA_CORPUS_ID ? parseInt(process.env.VECTARA_CORPUS_ID) : 0,
+    api_key: process.env.VECTARA_API_KEY || "",
+  });
+  const retriever = new VectaraRetriever(store);
+
+  // If you've uploaded the US constituion, this will return the relevant section
+  const docs = await retriever.getRelevantDocuments("Article 1 Section 7");
+
+  expect(docs.length).toBeGreaterThan(0);
+  // This checks that the document contains the relevant section
+  expect(docs[0].pageContent).toContain('All Bills for raising Revenue shall originate in the House of Representatives');
+
+  console.log(docs);
+});
+
+test("VectaraRetriever (without filter)", async () => {
```

Comment:
I'm having trouble spotting the differences between these tests. Can you explain how one is using a filter and the other isn't?

Reply:
I spoke with Ofer about this a couple weeks ago and he suggested testing once with a filter and once without. 

The test with the filter ensures that the results returned are actually relevant to the question. The test without the filter does not care about the result, it just ensures that something is outputted.

The test without the filter is a bit redundant in some way, should I remove it?